polaris-migrator/src/migrations/styles-insert-stylelint-disable/styles-insert-stylelint-disable.ts

```diff
@@ -0,0 +1,122 @@
+import type {FileInfo} from 'jscodeshift';
+import postcss, {Node, Comment, Plugin} from 'postcss';
+
+import stylelint from 'stylelint';
+
+const crossPlatformNewlineRegExp = /\r\n?|\n/;
+
+const plugin = (): Plugin => {
+  const polarisContextMsg =
+    '-- polaris: generated by polaris-migrator DO NOT COPY';
+
+  return {
+    postcssPlugin: 'insert-stylelint-disable',
+    Once(_, {result}) {
+      result.messages.forEach(({node, text: failureMsg, line, endLine}) => {
+        // If a polaris ignore comment exists above the node already,
+        // do nothing
+        if (node.prev()?.text?.includes(`-- polaris:`)) {
+          return;
+        }
+
+        // If the node is disable comment missing a polaris context message,
+        // append a generated context message
+        if (failureMsg.includes('-- polaris:')) {
+          node.text = `${node.text} ${polarisContextMsg}`;
+          return;
+        }
```

Comment:
this will never be true until [stylelint-polaris v5](https://github.com/Shopify/polaris/pull/7691/files) is shipped so I couldn't write a test for it. 

I could also omit the `if (failureMsg.includes('-- polaris:')` and assume all nodes that are comments that are failing need to be appended with the context string, then I could put the test back in